### DISCUSSION TITLE
How to use multiple canvas resources on a single screen?

### DISCUSSION THREAD

**Discussion by desktop-uidev** [AvaloniaUI/Avalonia#16631] [Q&A] [answered]
Hello!

I have an Icon in the resources. It is defined as a canvas since I didn't find any way to create `GeometryGroup` with different colors for nested geometries. Is it possible to use this icon multiple times in `ItemsControl.ItemTemplate` for example? When I try to do it application crashes with "`control already has visual parent`" exception. Is there no alternative to `x:Shared=false` from WPF?

`	<Canvas x:Key="Icon">
		<Path Fill="#FFEB5858" Data="M2 .992A1 1 0 0 1 2.995 0H12l6 6v12.993C18 19.55 17.555 20 17 20H3c-.552 0-1-.455-1-.992V.992z"/>
		<Rectangle Fill="#FFFFFFFF" Stroke="#FFB13B3B" Canvas.Left="-.5" Canvas.Top="7.5" Width="21" Height="11" RadiusX="1" RadiusY="1" />
		<Path Fill="#FF191E41" Data="M4.464 13.75V16H3.172V9.602h2.18c.632 0 1.137.196 1.514.588.376.393.564.903.564 1.53s-.186 1.122-.558 1.485c-.372.363-.888.545-1.547.545h-.861zm0-1.077h.888c.246 0 .436-.08.57-.241.136-.161.203-.396.203-.703 0-.32-.069-.574-.206-.763a.662.662 0 0 0-.554-.288h-.901v1.995zM8.392 16V9.602h1.691c.748 0 1.343.237 1.787.711.444.475.67 1.125.679 1.952v1.037c0 .84-.222 1.5-.666 1.98-.444.479-1.055.718-1.835.718H8.392zm1.292-5.322v4.25h.386c.431 0 .734-.114.91-.34.176-.228.268-.62.277-1.176V12.3c0-.598-.084-1.015-.25-1.25-.168-.236-.452-.36-.853-.372h-.47zm7.192 2.712h-2.008V16h-1.292V9.602h3.542v1.076h-2.25v1.64h2.008v1.072z"/>
		<Path Fill="#FFB13B3B" Data="M12 0l6 6h-5.01a.996.996 0 0 1-.99-.99V0z"/>
	`</Canvas>``

**Comment by IanRawley** [comment]
You can't do it with controls directly, but you can do it with Templates.  I similarly have a file full of icon resources like the following:
```
<DataTemplate x:Key="ERROR">
	<matav:MaterialIcon Kind="Error" Foreground="Red"></matav:MaterialIcon>
</DataTemplate>
```
which I then use by placing a basic ContentControl (or Button, etc as appropriate) and setting the ContentTemplate to the static resource.

**Reply by desktop-uidev** [reply]
Thank you, this definitely works. Not what I was hoping for though :))

**Comment by BAndysc** [comment, marked as answer]
Perfect timing :D https://github.com/AvaloniaUI/Avalonia/pull/16644 x:Shared PR
